kubeface/broadcast.py

```diff
@@ -0,0 +1,31 @@
+import logging
+import tempfile
+
+from . import common, naming, serialization, storage
+
+class Broadcast(object):
+
+    def __init__(self, storage_prefix, cache_key_prefix, data):
+        self.storage_prefix = storage_prefix
+        self.cache_key_prefix = cache_key_prefix
+        self.data = data
+
+    def compute_filename(self):
+        return naming.make_broadcast_name(self.cache_key_prefix)
+
+    """ Returns the path to which the data was written. """
+    def __getstate__(self):
+        # load data into a local tempfile and write out to storage
+        hash_path = '%s/%s' % (self.storage_prefix, self.compute_filename())
```

Comment:
nits:
* might call `hash_path` something different, since there is no hashing.
* I think getstate should return a dict: even though python doesn't require it I think that's how the pickle protocol is intended